### PR TITLE
Refactor: Use uuid.Must(uuid.NewRandom()) for lineage generation

### LINKED ISSUE #1525: Simplify Lineage uuid creation
Currently `internal/states/statemgr/lineage.go` only contains the `NewLineage` function that returns a UUID string.

This function is only called from `internal/states/statemgr/helper.go` and only once, so it can be simplified by directly calling UUID generation function `uuid.Must(uuid.NewRandom()).String()` to achieve the same result.

### PR DESCRIPTION
**Pull request by mikelolasagasti**
This commit removes the need for a separate lineage generation function by replacing it with a more concise and idiomatic approach using `uuid.Must(uuid.NewRandom())`.

Resolves #1525 

## Target Release

1.7.0

### CODE REVIEW

File: internal/states/statemgr/helper.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ import (
 // lineage identifier and serial 0, and returns a pointer to it.
 func NewStateFile() *statefile.File {
 	return &statefile.File{
-		Lineage:          NewLineage(),
```
[the remark this answers is not in the document]
Comment (mikelolasagasti) [reply internal/states/statemgr/helper.go:22]: The opaque string is part of `panic()` to notify that it was not possible to generate a UUID, but  `uuid.Must()` also panics in the same scenario.

[the remark this answers is not in the document]
Comment (cam72cam) [reply internal/states/statemgr/helper.go:22]: Is there a reason you didn't use https://pkg.go.dev/github.com/google/uuid#New directly?

**Comment by cam72cam** [comment]
If we are removing hashi's uuid library here, does it make sense to completely remove usage of the package?

File: internal/states/statemgr/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mikelolasagasti) [reply internal/states/statemgr/helper.go:22]: No, my mistake.

**Comment by mikelolasagasti** [comment]
I'm working to remove `github.com/hashicorp/go-uuid` but didn't manage to finish yesterday so I decided to push the changes that are more than just a "simple" replacement. I should have just opened the issue and add everything in a single PR.

**Comment by Evi1Pumpkin** [comment]
Hi @mikelolasagasti,
Thank you for your contribution. We truly appreciate the effort you put in. We kindly ask that for future reference, you take a moment to review our contribution guide (https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md#contributing-to-opentofu) to ensure you understand the process of opening a Pull Request in the OpenTofu project. 
Regrettably, we have decided not to accept this issue and contribution, as we are currently unable to see how this change benefits the OpenTofu users. We acknowledge that our process in this case was not ideal. We prematurely commented on this PR before deciding to accept this issue. We sincerely apologize for that.
Thank you sincerely for the time and effort you put into submitting this Pull Request!

**Pull request closed without merging**